### ISSUE TITLE
Provide Scarb package

### ISSUE THREAD

**Issue by mkaput** [OpenZeppelin/cairo-contracts#621]
We'd like to allow users to use OpenZeppelin contracts in Scarb-based projects. Primarily, this means all Protostar projects as it will soon use Scarb for dependency management exclusively.

**📝 Details**

Actually as far as I see, your `Makefile` flow could be easily swapped with Scarb-based solution, when Scarb will come with a test runner and starknet libfuncs verification solution. Both being on track for implementation soon. Am I right?

We also bootstrapped a commercial project that is kinda blocked on this.

**🔢 Code to reproduce bug**

A following Scarb.toml incantation should work:

```toml
[dependencies]
openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git" }
```

---

If you'd like, we can help with setting this up.

**Comment by martriay** [comment]
we for sure want to integrate with scarb (https://github.com/OpenZeppelin/cairo-contracts/issues/558) and move out from our Makefile soon.

that being said, we do not want to support, let alone encourage installing directly from git. we aim to release stable, installable packages that can be reviewed. git code is expected to evolve, be unstable, and even break sometimes.

**Comment by mkaput** [comment]
Okay, I will focus then on prepping Scarb (i.e. bundle cairo-test-runner & think on libfuncs whitelisting). I think 0.3.0-alpha.2 (i.e. the second next release) is a reasonable target. We can then talk on the process of migration :-)

> that being said, we do not want to support, let alone encourage installing directly from git. We aim to release stable, installable packages that can be reviewed. Git code is expected to evolve, be unstable, and even break sometimes. We also consider git installations dangerous since they can be target of malicious changes.

I do also share your opinion, but we decided with Starkware to stick with Git for the time being for operational simplicity. In the long-term future, we definitely want to create a way of hosting packages in immutable way, i.e. like crates.io; but this is one of the lower priority things as for now.

There is one potential not-ideal solution for this problem that is already doable in Scarb: you can pin dependencies to specific commit hash (via `rev` field). Commit hashes basically guarantee immutability of code, because even the slightest evil change will not generate the same hash (sans SHA-1 collisions). The only problem is that the entire ecosystem would have to do synchronized upgrade steps, but I think we can hope that this won't end up being a big deal for first few months?

**Comment by mkaput** [comment]
Update, Scarb 0.3 is gonna be released today/tomorrow because Cairo 1.1.0 was released. Obviously we didn't manage to bring all we need for now, so Scarb 0.4 or so is the goal now.

**Comment by mkaput** [comment]
Scarb 0.4 is out now and should support everything needed for this project: https://github.com/software-mansion/scarb/releases/tag/v0.4.0

@maciektr will reach out to you soon with a helping hand 😃

**Comment by amanusk** [comment]
Can a possible solution here be either:
* Specifying a commit/tag when adding a scarb dependency
and/or
* Having a "scarb.lock" file that sets the git commits on first build?

**Comment by martriay** [comment]
> Can a possible solution here be either:
> 
> * Specifying a commit/tag when adding a scarb dependency
>   and/or
> * Having a "scarb.lock" file that sets the git commits on first build?

Tags would do for the time being. These proposals require changes in scarb or is the tool ready?

**Comment by martriay** [comment]
Resolved by #671
